src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -37,66 +37,69 @@ internal partial class WebSocketsTransport : ITransport
+                        _webSocket.Options.Cookies = httpConnectionOptions.Cookies;
+                    }
 
-                if (httpConnectionOptions.ClientCertificates != null)
-                {
-                    _webSocket.Options.ClientCertificates.AddRange(httpConnectionOptions.ClientCertificates);
-                }
+                    if (httpConnectionOptions.ClientCertificates != null)
+                    {
+                        _webSocket.Options.ClientCertificates.AddRange(httpConnectionOptions.ClientCertificates);
+                    }
 
-                if (httpConnectionOptions.Credentials != null)
-                {
-                    _webSocket.Options.Credentials = httpConnectionOptions.Credentials;
-                }
+                    if (httpConnectionOptions.Credentials != null)
+                    {
+                        _webSocket.Options.Credentials = httpConnectionOptions.Credentials;
+                    }
 
-                if (httpConnectionOptions.Proxy != null)
-                {
-                    _webSocket.Options.Proxy = httpConnectionOptions.Proxy;
-                }
+                    if (httpConnectionOptions.Proxy != null)
+                    {
+                        _webSocket.Options.Proxy = httpConnectionOptions.Proxy;
+                    }
 
-                if (httpConnectionOptions.UseDefaultCredentials != null)
-                {
-                    _webSocket.Options.UseDefaultCredentials = httpConnectionOptions.UseDefaultCredentials.Value;
+                    if (httpConnectionOptions.UseDefaultCredentials != null)
+                    {
+                        _webSocket.Options.UseDefaultCredentials = httpConnectionOptions.UseDefaultCredentials.Value;
+                    }
                 }
 
-                httpConnectionOptions.WebSocketConfiguration?.Invoke(_webSocket.Options);
-
-                _closeTimeout = httpConnectionOptions.CloseTimeout;
+                // Set this header so the server auth middleware will set an Unauthorized instead of Redirect status code
+                // See: https://github.com/aspnet/Security/blob/ff9f145a8e89c9756ea12ff10c6d47f2f7eb345f/src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieAuthenticationEvents.cs#L42
+                _webSocket.Options.SetRequestHeader("X-Requested-With", "XMLHttpRequest");
             }
 
-            // Set this header so the server auth middleware will set an Unauthorized instead of Redirect status code
-            // See: https://github.com/aspnet/Security/blob/ff9f145a8e89c9756ea12ff10c6d47f2f7eb345f/src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieAuthenticationEvents.cs#L42
-            _webSocket.Options.SetRequestHeader("X-Requested-With", "XMLHttpRequest");
+            // Allow this callback to run in the browser in case it has code that will not throw PNSE
+            httpConnectionOptions.WebSocketConfiguration?.Invoke(_webSocket.Options);
```

Comment:
Should we do this inside the if block? None of the SignalR defaults are configured, even if the options could be configured, they would be working with values that are not initialized.

Reply:
>            // Allow this callback to run in the browser in case it has code that will not throw PNSE

We can move it inside the check or not